modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java

```diff
@@ -0,0 +1,203 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.integration.test.idp.mgt;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.IdentityProviderProperty;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.identity.integration.common.clients.Idp.IdentityProviderMgtServiceClient;
+import org.wso2.identity.integration.test.oauth2.OAuth2ServiceAbstractIntegrationTest;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.IOException;
+
+/**
+ * This test class is to test Preference API integration with Login & Recovery Portals.
+ */
+public class PreferenceAPIIntegrationUITestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String ENABLE_SELF_REGISTRATION_PROP_KEY = "SelfRegistration.Enable";
+    private static final String ENABLE_USERNAME_RECOVERY_PROP_KEY = "Recovery.Notification.Username.Enable";
+    private static final String ENABLE_PASSWORD_QS_RECOVERY_PROP_KEY = "Recovery.Question.Password.Enable";
+    private static final String ENABLE_PASSWORD_NOTIFICATION_RECOVERY_PROP_KEY =
+            "Recovery.Notification.Password.Enable";
+    private static final String CALLBACK_URL = "https://localhost/callback";
+    private static final String RECOVERY_USERNAME_CONTENT = "id=\"usernameRecoverLink\"";
+    private static final String RECOVERY_PASSWORD_CONTENT = "id=\"passwordRecoverLink\"";
+    private static final String RECOVERY_ENDPOINT_QS_CONTENT = "name=\"recoveryOption\" value=\"SECURITY_QUESTIONS\"";
+    private static final String RECOVERY_ENDPOINT_NOTIFICATION_CONTENT = "name=\"recoveryOption\" value=\"EMAIL\"";
+    private static final String CREATE_ACCOUNT_CONTENT = "id=\"registerLink\"";
+    private static final String RECOVERY_ENDPOINT_URL =
+            "https://localhost:9853/accountrecoveryendpoint/recoveraccountrouter.do";
+
+    private IdentityProvider superTenantResidentIDP;
+    private IdentityProviderMgtServiceClient superTenantIDPMgtClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
```

Comment:
Using a data provider can't we run the test for a tenant too?

Reply:
Yes we can add a tenant using a data provider